### PR TITLE
Fix NASNet input shape computation

### LINKED ISSUE #16791: NASNet error using custom input shape without top layer
https://github.com/keras-team/keras/blob/07e13740fd181fc3ddec7d9a594d8a08666645f6/keras/applications/nasnet.py#L174-L180

Shouldn't the parameter `require_flatten` uses `include_top` value instead of constant `True`? 
Using `True` makes custom shaped input invalid when using size that isn't the `default_size` with `include_top = False`
Checked several other models such as efficientnet_v2 and xception, they all use `include_top` as value instead of `True`

### PR DESCRIPTION
**Pull request by ianjjohnson**
Fixes #16791.

Thank you to [ryukinlika](https://github.com/ryukinlika) who found the issue and identified an appropriate fix.

### CODE REVIEW

**Comment by ianjjohnson** [comment]
@[qlzh727](https://github.com/qlzh727)

**Comment by qlzh727** [comment]
Can u update the unit test Ian?

File: keras/applications/nasnet.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def NASNet(
         default_size=default_size,
         min_size=32,
         data_format=backend.image_data_format(),
-        require_flatten=True,
+        require_flatten=include_top,
```
Comment (qlzh727) [inline keras/applications/nasnet.py:191]: Please add/update unit test for this.

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: keras/applications/nasnet.py (hunk 1 above)
Comment (ianjjohnson) [reply keras/applications/nasnet.py:191]: Added a unit test for this case and verified that test passes locally (and fails without the change).

**Comment by qlzh727** [comment]
Ian, it seems that the presubmit test is not auto triggering since you didn't accept the invite we sent to join the keras-team github org. Could you check that? (which should allow u more freely test the PR.)

**Comment by ianjjohnson** [comment]
> Ian, it seems that the presubmit test is not auto triggering since you didn't accept the invite we sent to join the keras-team github org. Could you check that? (which should allow u more freely test the PR.)

I've accepted the invite -- it looks like this didn't retroactively trigger the presubmits, but hopefully it will next time?

**Review by qlzh727 — APPROVED** [review]
Thanks for the fix.

**Pull request merged**